docs/proposals/storage-capacity-proposal.md

```diff
@@ -0,0 +1,132 @@
+# Proposal: Implement storage capacity feature
+
+Last updated: 08.07.2022
+
+## Abstract
+
+Specify required changes to support storage capacity feature, compare storage tracking with ACR and CSIStorageCapacity resources.
+
+## Background
+
+Currently the Plugin uses AC and ACR CRD algoritm for dynamic provisioning. The scheduler-extender (SE) sidecar searches all PVCs, creates corresponding CapacityRequests and requests reservations by creating ACR objects. List of reserved ACRs is filled by reservationcontroller within csi driver controller.
+This algorithm has some disadvantages:
+- Every user of plugin must use the Extender
+- Reservation blocks suitable AC on every requested node. It slows scheduling, because reserved ACs can't be used for other volumes until ACR is deleted
+- Reservation cannot be made without pod passing scheduling cycle, which makes at least two scheduling cycles required to place a pod  
+
+## Proposal
+
+Implement GetCapacity, support Storage Capacity Feature, make an extra option for users to remove Extender and ACR resource.
+
+## Rationale
+
+Advantages:
+- First of all, it is a step towards standartization of storage tracking algorithm
+- Implementing Storage Capacity feature may speed up scheduling
+- Generic ephemeral volumes are already supported by driver, they can be provisioned using storage capacity tracking too
+- Storage Capacity objects are being produced and updated independently from PVC. This means that when the pod first goes through the scheduling cycle, it is likely that information about available capacity have already been gathered to Storage Capacity objects and the
+pod can be scheduled at the same cycle
+
+Disadvantages:
+- Storage Capacity objects require more memory in etcd. In contrast, ACR are deleted after volume creation
+- The feature doesn't solve races issues, two pods still may reserve same disk
```

Comment:
Will pod stuck in pending state in this case?

Reply:
Honestly, I don't know how this happens, @safronovD have mentioned this issue, but he's busy now

Reply:
Or user/Operator should remove PVC and delete pod to restart scheduling process as workaround